index.js

```diff
@@ -125,7 +125,10 @@ Discovery.prototype.destroy = function (cb) {
     self.tracker.removeListener('peer', self._onTrackerPeer)
     self.tracker.removeListener('update', self._onTrackerAnnounce)
     tasks.push(function (cb) {
-      self.tracker.destroy(cb)
+      process.nextTick(function () {
+        self.tracker.destroy(cb)
```

Comment:
Right, but the UDP Tracker calls it's own socket.close() method in cleanup();

self.tracker = null is just to empty the pointer.

I experimented last night and it works. That's why I submitted a PR for the cleanup time update for bittorrent-tracker.

TL;DR : line 122 of this code you send a 'close()' method. This method actually has it's own cleanup implemented.
Trust me, the script runs and turns off. Do you want me to submit a test code that works with logs as I described?